server/src/main/java/io/spine/server/command/CommandSubstMethod.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.command;
+
+import io.spine.annotation.Internal;
+import io.spine.base.ThrowableMessage;
+import io.spine.core.CommandClass;
+import io.spine.core.CommandContext;
+import io.spine.server.model.AbstractHandlerMethod;
+import io.spine.server.model.MethodAccessChecker;
+import io.spine.server.model.MethodExceptionChecker;
+
+import java.lang.reflect.Method;
+import java.util.function.Predicate;
+
+/**
+ * A method that produces one or more command messages in response to an incoming command.
+ *
+ * @author Alexander Yevsyukov
+ */
+@Internal
+public final class CommandSubstMethod
```

Comment:
Please use the whole word. `Command Substance Method` is what comes into my head first.

Reply:
`subst` is an accepted shortening. This API isn't end user, and I don't expect much of a confusion anyway. I'm going to visit this issue once again when adding more command generation methods.